helpers/metadata.go

```diff
@@ -33,3 +38,37 @@ func HasLabelWithValue(object client.Object, label, value string) bool {
 
 	return false
 }
+
+// CopyLabelsByPrefix copies all labels from a source object to a destination object where the key matches the specified prefix.
+// If replacementPrefix is different from prefix, the prefix will be replaced while performing the copy.
+func CopyLabelsByPrefix(src *metav1.ObjectMeta, dest *metav1.ObjectMeta, prefix string, replacementPrefix string) {
+	if dest.Labels == nil {
+		dest.Labels = make(map[string]string)
```

Comment:
Maybe consider using the object's Get & Set methods rather than going directly.
Something like this...
```
if dest.GetLabels() == nil {
		dest.SetLabels(map[string]string{})
```
Note: please consider this an _**optional**_ suggestion, because it's highly pedantic, and purely subjective. Accessing exported maps is fine, even if the maps have getter's and setter's.

Reply:
I applied your suggestion and added an extra `nil` check to each helper in the process.